.github/workflows/build-test-deploy.yml

```diff
@@ -340,13 +341,22 @@ jobs:
           installCommand: |
             nix-channel --add https://nixos.org/channels/nixpkgs-22.05-darwin nixpkgs
             nix-channel --update
-            nix-env -iA nixpkgs.cachix
+            nix-env -iA nixpkgs.cachix nixpkgs.docker nixpkgs.docker-buildx
           authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
           name: composable-community
+      - uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
       - run: |
-          # TODO: upload docker image
-          # Only on trigger by user?
-          nix build .#devnet-container --no-update-lock-file --show-trace -L
+          # enable required features (see https://github.com/cachix/install-nix-action/issues/19)
+          mkdir -p ~/.config/nix
+          echo "system-features = kvm" >> ~/.config/nix/nix.conf
+      - run: |
+          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#devnet-container --no-update-lock-file --show-trace -L
+          docker load --input ./result
+          docker tag composable-devnet-container:latest "${{ env.DOCKER_REGISTRY_NAME }}/composable-sandbox:latest"
```

Comment:
imho retag as latest should happen only after all tests passed, and iff merge to main, like devnet.

i mean we should not replace latest with arbitrary branch

Reply:
yes, this job will be run only from main branch (see https://github.com/ComposableFi/composable/pull/1494/files#diff-22f49dea8896855b65ec7f98c2dccafa1b7317f851952fc98ee59f29d84f8af5R312)

Reply:
also, this job will be run after all tests are passed.